src/librustc_driver/lib.rs

```diff
@@ -378,6 +380,64 @@ pub trait CompilerCalls<'a> {
     // Create a CompilController struct for controlling the behaviour of
     // compilation.
     fn build_controller(&mut self, _: &Session, _: &getopts::Matches) -> CompileController<'a>;
+
+    fn mir_passes(&mut self) -> Rc<Passes>{
```

Comment:
This seems like a reasonable approach, but if we're going to bother with this, then we should introduce a method (or free function) like `fn default_mir_passes() -> Passes`, and make this default implementation just do `fn mir_passes(&mut self) -> Rc<Passes> { Rc::new(self.default_mir_passes()) }` or whatever.